mw-webapp/src/hooks/useLoad.ts

```diff
@@ -0,0 +1,75 @@
+import {useEffect, useState} from "react";
+
+/**
+ * Use Load hook Props
+ */
+interface useLoadProps<T, D> {
+
+  /**
+   * Callback that is called to fetch data
+   */
+  loadData: () => Promise<T>;
+
+  /**
+   * Callback that is called to validate data
+   */
+  validateData?: (data: T) => boolean;
+
+  /**
+   * Callback that is called on fetch and validation success
+   */
+  onSuccess: (data: T) => void;
+
+  /**
+   * Callback this is called on fetch or validation error
+   */
+  onError?: (error: Error) => void;
+
+  /**
+   * Passed dependency to refetch and revalidate data
+   */
+  dependency?: D;
```

Comment:
Should dependency be kind of react state? If yes, probably it will be better to have a list of dependencies, agree?

Reply:
You mean something like `dependencies?: any[]`; and pass it to useEffect?

Reply:
Refactored